docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,999 @@
+    "qi_training = QuantumInstance(backend, shots=batch_size)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Then we create a quantum instance for the evaluation, we choose a higher number of shots to get better insights."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "qi_sampling = QuantumInstance(backend, shots=10000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Initialization of quantum generator and the respective gradient\n",
+    "Next, we define a function that creates the quantum generator from a given parameterized quantum circuit. As parameters this function also takes a quantum instance to be used for data sampling. We wrap a created quantum neural network in `TorchConnector` to make use of PyTorch training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def create_generator(qc, quantum_instance) -> TorchConnector:\n",
+    "    circuit_qnn = CircuitQNN(\n",
+    "        qc,\n",
+    "        input_params=[],\n",
+    "        weight_params=qc.parameters,\n",
+    "        quantum_instance=quantum_instance,\n",
+    "        sampling=True,\n",
+    "        sparse=False,\n",
+    "        input_gradients=True,  # TODO: !!!!!! Check out if they are required !!!!!!!\n",
+    "        interpret=lambda x: grid_elements[\n",
+    "            x\n",
+    "        ],  # TODO: !!!!!!!!!!! remove access to the global variable!!!!!\n",
+    "    )\n",
+    "\n",
+    "    return TorchConnector(circuit_qnn)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "We also define a function that calculates the gradients for the quantum generator. Depending on the backend, the gradients may be returned in a sparse format. Since PyTorch provides only a limited support for sparse gradients, we need to write a custom function for gradient based training.\n",
+    "\n",
+    "This function takes a quantum circuit that represents our quantum neural network, a list of parameters to evaluate the gradients with respect to, and a list of parameter values. The function returns a list of gradients for the sampling probabilities of the quantum neural network."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def generator_grad(qc, parameters, parameter_values) -> Iterable:\n",
+    "    # TODO: !!!!!!!! remove access to the global variable: backend=qi_training !!!!!!!\n",
```

Comment:
Just a reminder that this TODO is also missing

Reply:
Yes, this TODO is fixed.